components/script/script_module.rs

```diff
@@ -880,6 +940,74 @@ impl ModuleOwner {
             },
         }
     }
+
+    #[allow(unsafe_code)]
+    /// <https://html.spec.whatwg.org/multipage/#hostimportmoduledynamically(referencingscriptormodule,-specifier,-promisecapability):fetch-an-import()-module-script-graph>
+    /// Step 6-9
+    pub fn finish_dynamic_module(
+        &self,
+        module_identity: ModuleIdentity,
+        dynamic_module_id: DynamicModuleId,
+    ) {
+        let global = self.global();
+
+        let module = global.dynamic_module_list().remove(dynamic_module_id);
+
+        let cx = global.get_cx();
+        let module_tree = module_identity.get_module_tree(&global);
+
+        // In the timing of executing this `finish_dynamic_module` function,
+        // we've run `find_first_parse_error` which means we've had the highest
+        // priority error in the tree. So, we can just get both `network_error` and
+        // `rethrow_error` directly here.
+        let network_error = module_tree.get_network_error().borrow().as_ref().cloned();
+        let existing_rethrow_error = module_tree.get_rethrow_error().borrow().as_ref().cloned();
+
+        let execution_err = if network_error.is_none() && existing_rethrow_error.is_none() {
+            if let Some(ref record) = &*module_tree.get_record().borrow() {
+                let evaluated = module_tree.execute_module(&global, record.handle()).err();
```

Comment:
I think it might be risky to keep the borrow active while we call execute_module. We should clone the record instead.

Reply:
Indeed! Let me fix it! Thanks!

Reply:
Fixed in fe0462d